### PR TITLE
Return rejected promise on error

### PR DESCRIPTION
**Pull request by dsblv**
Just a friendly PR to keep things up-to-date.

I assumed adding `pinkie-promise` to dependencies is not that bad, as `got` already depends on it.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 'use strict';
 var got = require('got');
 var objectAssign = require('object-assign');
+var PinkiePromise = require('pinkie-promise');
 
 function ghGot(path, opts) {
 	if (!path) {
-		throw new Error('path required');
+		return PinkiePromise.reject(new Error('path required'));
```
Comment (shinnn) [inline index.js]: Not directly related to this PR, but this error should be `TypeError` IMHO.

Comment (dsblv) [inline index.js]: @shinnn indeed

File: index.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 'use strict';
 var got = require('got');
 var objectAssign = require('object-assign');
+var PinkiePromise = require('pinkie-promise');
```
Comment (sindresorhus) [inline index.js]:
```js
var Promise = require('pinkie-promise');
```

Makes it easier to remove pinkie-promise when polyfill is no longer needed.

**Comment by dsblv** [comment]
:sparkles:

**Pull request closed without merging**